drivers/platform/stm32/stm32_i2c.c

```diff
@@ -70,9 +70,29 @@ int32_t stm32_i2c_init(struct i2c_desc **desc,
 	}
 
 	descriptor->extra = xdesc;
-	xinit = param->extra;
 
-	xdesc->hi2c.Instance = xinit->base;
+	switch (param->device_id) {
+#if defined(I2C1)
+	case 1:
+		base = I2C1;
+		break;
+#endif
+#if defined(I2C2)
+	case 2:
+		base = I2C2;
+		break;
+#endif
+#if defined(I2C3)
+	case 3:
+		base = I2C3;
+		break;
+#endif
+	default:
+		ret = -EINVAL;
+		goto error_2;
+	};
+
+	xdesc->hi2c.Instance = base;
```

Comment:
nitpick: I'd say there is no need for the intermediate pointer 'base'. We can just assign 'xdesc->hi2c.Instance' in switch.